### PR TITLE
Cluster-api config option and dashboard

### PR DESCRIPTION
**Pull request by viktor-f**
**What this PR does / why we need it**:
Removed calico-node from the tests since it is not managed by apps. It felt odd that they were the only pods tested that are installed by kubespray/cluster-api.
Added `global.clusterApi` option to enable changes needed for cluster-api managed clusters.
Added dashboard for cluster-api management.

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

cluster-api-dashboard

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/charts/grafana-ops/dashboards/cluster-api-dashboard.json [hunk 1]
```diff
@@ -0,0 +1,1156 @@
...
+          "text": [
+            "All"
+          ],
+          "value": [
+            "$__all"
+          ]
+        },
+        "datasource": {
+          "type": "prometheus",
+          "uid": "PE090269EBE049DA8"
```
Comment (lunkan93) [inline helmfile/charts/grafana-ops/dashboards/cluster-api-dashboard.json:1126]:
```suggestion
          "uid": "${datasource}"
```

File: WIP-CHANGELOG.md [hunk 2]
```diff
@@ -7,9 +7,15 @@
 - Add additional pre-commit checks with codespell, editorconfig, and shellcheck
 - Added exceptions for falco alerts generated by Tekton
 - Added exceptions for falco alerts generated by ingress-nginx
+- Added config option to enable changes needed for cluster-api managed clusters
```
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:10]:
```suggestion
- Added config option to support [Cluster API](https://cluster-api.sigs.k8s.io/) managed Kubernetes clusters
```

File: helmfile/charts/calico-accountant/templates/rbac.yaml [hunk 3]
```diff
@@ -1,14 +1,13 @@
----
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: calico-accountant
+# ---
+# apiVersion: v1
+# kind: ServiceAccount
+# metadata:
+#   name: calico-accountant
```
Comment (aarnq) [inline helmfile/charts/calico-accountant/templates/rbac.yaml:5]: If it's not going to use this SA, and instead use the calico-node then why do we have the rest of this rbac? (Also any reason why it doesn't use it's own SA?)

File: helmfile/charts/grafana-ops/dashboards/cluster-api-dashboard.json [hunk 4]
```diff
@@ -0,0 +1,1156 @@
...
+          "exemplar": false,
+          "expr": "capi_machine_status_condition{cluster_name=~\"$cluster\", name!~\".*control-plane.*\", type=\"Ready\"} == 1",
+          "format": "table",
+          "instant": true,
+          "legendFormat": "{{name}}",
+          "range": false,
+          "refId": "A"
+        }
+      ],
+      "title": "Control plane replicas",
```
Comment (aarnq) [inline helmfile/charts/grafana-ops/dashboards/cluster-api-dashboard.json:1035]: I assume this should be MachineDeployment replicas?

File: helmfile/charts/calico-accountant/templates/rbac.yaml (hunk 3 above)
Comment (Xartos) [reply helmfile/charts/calico-accountant/templates/rbac.yaml:5]: Yea, if it's not used then I'd say that. We can always dig it back again if we want to get it back.

**Review by Xartos — APPROVED** [review]

File: helmfile/charts/calico-accountant/templates/rbac.yaml (hunk 3 above)
Comment (viktor-f) [reply helmfile/charts/calico-accountant/templates/rbac.yaml]: Removed the file now and created an issue for fixing these permissions later.
https://github.com/elastisys/compliantkubernetes-apps/issues/1624

**Review by lunkan93 — APPROVED** [review]

**Pull request merged**